packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -422,15 +434,18 @@ If you provided a React element for the optionText prop, you must also provide t
         newInputValue: string,
         reason: string
     ) => {
-        if (!doesQueryMatchSelection(newInputValue, event?.type)) {
+        if (
+            event?.type === 'change' ||
+            !doesQueryMatchSelection(newInputValue, event?.type)
```

Comment:
We no longer need to check the `event` type in `doesQueryMatchSelection()`, because it's already done in `handleInputChange()`.

`handleInputChange()` was the only place where  `doesQueryMatchSelection()` was called with a second parameter, so we can safely remove it IMO.